src/LibYear.Lib/FileTypes/XmlProject.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace LibYear.Lib.FileTypes
+{
+    public abstract class XmlProject : IProjectFile
+    {
+        public virtual string FileName => throw new NotSupportedException();

```

Comment:
Group by `private`, then `protected` then `public`. Easier to read and maintain.